### DISCUSSION TITLE
How to get the image status in viewmodel

### DISCUSSION THREAD

**Discussion by howzew** [sn4k3/UVtools#951] [Q&A] [answered]
In my business, I need to perform some operations on the image, such as replacement, in the MVVM mode, I modify the image in the viewmodel, and then bind it to the control. I zoom in/out the image before replacing, but when I replaced, I found that the new image didn't keep the zoom in/out, but in WPF it kept the zoom in/out, and I guess the control seemed to be reset when the image property method changed, so if I wanted to keep the zooming effect, how do I get this state in the viewmodel and apply it to the new image?

At the same time, I found another interesting thing, for example, I want to change the black in the image to white, I wrote a `changecolor `method, when I use this method, the image does not change in any way, the black does not change to white, but when I zoom in or out of the image, I find that the black in the image changes to white, that is, the image has actually been modified, but the control seems to need to be refreshed before it can be displayed.

These two phenomena confuse me and seem to be the opposite.

**Comment by sn4k3** [comment, marked as answer]
> when I replaced, I found that the new image didn't keep the zoom in/out

It does by default, but if you using `AutoZoomToFit` or `ConstrainZoomOutToFitLevel` to `true` it's expected to affect zoom after replace image.

>  I want to change the black in the image to white, I wrote a changecolor method, when I use this method, the image does not change in any way.

Because any change to image pixels will not trigger a render. After your modifications you must call `LayerImage.TriggerRender()` or `LayerImage.InvalidateVisual()`

> when I zoom in or out of the image, I find that the black in the image changes to white, that is, the image has actually been modified

Zoom change always re-render the control.

**Reply by howzew** [reply]
Thanks for solving my problem
